packages/edit-site/src/components/global-styles/screen-revisions/index.js

```diff
@@ -55,13 +55,37 @@ function ScreenRevisions() {
 	const { setEditorCanvasContainerView } = unlock(
 		useDispatch( editSiteStore )
 	);
+	const selectedRevisionMatchesEditorStyles = areGlobalStyleConfigsEqual(
+		globalStylesRevision,
+		userConfig
+	);
+	const shouldSelectFirstItem =
+		! selectedRevisionMatchesEditorStyles &&
+		( ! selectedRevisionId || 'unsaved' === selectedRevisionId );
 
 	useEffect( () => {
 		if ( editorCanvasContainerView !== 'global-styles-revisions' ) {
 			goBack();
 			setEditorCanvasContainerView( editorCanvasContainerView );
+			return;
+		}
+
+		/*
+			Ensure that the first item is selected where no revision is selected
+			and the selected styles don't match the current editor styles.
+			This is required in case editor styles are changed outside the revisions panel,
+			e.g., via the reset styles function of useGlobalStylesReset().
+			See: https://github.com/WordPress/gutenberg/issues/55866
+		*/
+		if ( shouldSelectFirstItem ) {
+			setGlobalStylesRevision( revisions[ 0 ] );
 		}
-	}, [ editorCanvasContainerView ] );
+	}, [
+		editorCanvasContainerView,
+		shouldSelectFirstItem,
+		revisions[ 0 ],
+		setGlobalStylesRevision,
+	] );
```

Comment:
This dependency array is complaining about some missing dependencies:

<img width="792" alt="image" src="https://github.com/WordPress/gutenberg/assets/14988353/46a2a67c-12c9-4f39-a680-213a5cfdba6d">

To neaten things up, would it work to split out this new behaviour to a separate `useEffect`, that way the dependencies related to the `editorCanvasConatinerView` can be treated separately?

Reply:
Totally. I'll split them up to be neater.